deps/rabbit_common/src/rabbit_env.erl

```diff
@@ -1718,14 +1718,14 @@ parse_conf_env_file_output2([Line | Lines], Vars) ->
     end.
 
 is_sh_set_x_output(Line) ->
-    re:run(Line, "^\\++ ", [{capture, none}]) =:= match.
+    re:run(unicode:characters_to_binary(Line), "^\\++ ", [{capture, none}]) =:= match.
 
 is_sh_function(_, []) ->
     false;
 is_sh_function(Line, Lines) ->
-    re:run(Line, "\\s\\(\\)\\s*$", [{capture, none}]) =:= match
+    re:run(unicode:characters_to_binary(Line), "\\s\\(\\)\\s*$", [{capture, none}]) =:= match
     andalso
-    re:run(hd(Lines), "^\\s*\\{\\s*$", [{capture, none}]) =:= match.
+    re:run(unicode:characters_to_binary(hd(Lines)), "^\\s*\\{\\s*$", [{capture, none}]) =:= match.
```

Comment:
Likewise for this chunk. For example:
```erlang
    UnicodeLine1 = unicode:characters_to_binary(Line),
    UnicodeLine2 = unicode:characters_to_binary(hd(Lines)),
    re:run(UnicodeLine1, "\\s\\(\\)\\s*$", [{capture, none}]) =:= match
    andalso
    re:run(UnicodeLine2, "^\\s*\\{\\s*$", [{capture, none}]) =:= match.
```

Reply:
Ended up using this approach. Not perfect because we compute UnicodeLine2 outside of short-circuit, but the rest of the rabbit_env with functions like parse_unquoted_literal and rabbit_env_SUITE expects erlang strings. Too big rewrite to do.